File: apps/android/app/src/main/java/com/retailmanagement/data/owner/VendorReviewContracts.kt
package com.retailmanagement.data.owner

import com.google.gson.annotations.SerializedName

data class VendorReviewOrderRecord(
    @SerializedName("order_number") val orderNumber: String,
    @SerializedName("order_date") val orderDate: String,
    @SerializedName("supplier_id") val supplierId: String,
    @SerializedName("supplier_name") val supplierName: String,
    val currency: String,
    @SerializedName("source_document_total_amount") val sourceDocumentTotalAmount: Double,
    @SerializedName("document_payment_status") val documentPaymentStatus: String,
    @SerializedName("item_reconciliation_status") val itemReconciliationStatus: String?,
    @SerializedName("line_items") val lineItems: List<VendorReviewLineItem>
)

data class VendorReviewLineItem(
    @SerializedName("source_line_number") val sourceLineNumber: Int,
    @SerializedName("supplier_item_code") val supplierItemCode: String?,
    @SerializedName("unit_cost_cny") val unitCostCny: Double?,
    val quantity: Int?,
    @SerializedName("line_total_cny") val lineTotalCny: Double?,
    val size: String?,
    @SerializedName("material_description") val materialDescription: String?,
    @SerializedName("display_name") val displayName: String?
)

enum class ReviewLineStatus {
    @SerializedName("unreviewed") UNREVIEWED,
    @SerializedName("verified") VERIFIED,
    @SerializedName("needs_follow_up") NEEDS_FOLLOW_UP
}

data class ReviewLineState(
    var status: ReviewLineStatus,
    var note: String,
    var targetSkuId: String,
    var updatedAt: Long?
)

data class ReviewOrderState(
    var lines: MutableMap<String, ReviewLineState>
)

data class SupplierReviewWorkspaceState(
    var schemaVersion: Int,
    var supplierId: String,
    var savedAt: Long?,
    var orders: MutableMap<String, ReviewOrderState>
)

File: apps/ios/retailmanagement/Models/VendorReviewContracts.swift
//
//  VendorReviewContracts.swift
//  retailmanagement
//
//  Data models for Supplier Invoice/OCR Review.
//

import Foundation

struct VendorReviewOrderRecord: Codable, Identifiable {
    var id: String { orderNumber }
    let orderNumber: String
    let orderDate: String
    let supplierId: String
    let supplierName: String
    let currency: String
    let sourceDocumentTotalAmount: Double
    let documentPaymentStatus: String
    let itemReconciliationStatus: String?
    let lineItems: [VendorReviewLineItem]

    enum CodingKeys: String, CodingKey {
        case orderNumber = "order_number"
        case orderDate = "order_date"
        case supplierId = "supplier_id"
        case supplierName = "supplier_name"
        case currency
        case sourceDocumentTotalAmount = "source_document_total_amount"
        case documentPaymentStatus = "document_payment_status"
        case itemReconciliationStatus = "item_reconciliation_status"
        case lineItems = "line_items"
    }
}

struct VendorReviewLineItem: Codable, Identifiable {
    var id: String { "\(sourceLineNumber)" }
    let sourceLineNumber: Int
    let supplierItemCode: String?
    let unitCostCny: Double?
    let quantity: Int?
    let lineTotalCny: Double?
    let size: String?
    let materialDescription: String?
    let displayName: String?

    enum CodingKeys: String, CodingKey {
        case sourceLineNumber = "source_line_number"
        case supplierItemCode = "supplier_item_code"
        case unitCostCny = "unit_cost_cny"
        case quantity
        case lineTotalCny = "line_total_cny"
        case size
        case materialDescription = "material_description"
        case displayName = "display_name"
    }
}

// MARK: - Local Workspace State (Persisted)

enum ReviewLineStatus: String, Codable {
    case unreviewed
    case verified
    case needsFollowUp = "needs_follow_up"
}

struct ReviewLineState: Codable {
    var status: ReviewLineStatus
    var note: String
    var targetSkuId: String
    var updatedAt: Date?
}

struct ReviewOrderState: Codable {
    var lines: [String: ReviewLineState]
}

struct SupplierReviewWorkspaceState: Codable {
    var schemaVersion: Int
    var supplierId: String
    var savedAt: Date?
    var orders: [String: ReviewOrderState]
}
